Sources/SWBUtil/ProcessInfo.swift

```diff
@@ -167,6 +227,110 @@ public enum OperatingSystem: Hashable, Sendable {
             return .elf
         }
     }
+
+    private func detectHostLinuxDistribution() -> LinuxDistribution? {
+        return detectHostLinuxDistribution(fs: localFS)
+    }
+
+    /// Detects the Linux distribution by examining system files with an injected filesystem
+    /// Start with the "generic" /etc/os-release then fallback
+    /// to various distribution named files.
+    public func detectHostLinuxDistribution(fs: any FSProxy) -> LinuxDistribution? {
+        // Try /etc/os-release first (standard)
+        let osReleasePath = Path("/etc/os-release")
+        if fs.exists(osReleasePath) {
+            if let osReleaseData = try? fs.read(osReleasePath),
+               let osRelease = String(data: Data(osReleaseData.bytes), encoding: .utf8) {
```

Comment:
nit: prefer the stdlib constructor instead of the Foundation one:

```suggestion
               let osRelease = String(decoding: Data(osReleaseData.bytes), as: UTF8.self) {
```